File: 1930_unique_length_3_palindromic_subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string y) {
        int z = y.size(), x = 0;
        vector<int> p(26, -1), q(26, z);
        for (int i = 0; i < z; ++i) {
            if (p[y[i] - 'a'] == -1) p[y[i] - 'a'] = i;
            q[y[i] - 'a'] = i;
        }
        for (int o = 0; o < 26; ++o) {
            if (p[o] != -1 && q[o] != z) {
                unordered_set<int> r;
                for (int t = p[o] + 1; t < q[o]; ++t) r.insert(y[t]);
                x += r.size();
            }
        }
        return x;
    }
};

File: 1930_unique_length_3_palindromic_subsequences.py
class Solution(object):
    def countPalindromicSubsequence(self, d):
        sz, ans = len(d), 0
        left, right = [-1] * 26, [sz] * 26
        for i, ch in enumerate(d):
            idx = ord(ch) - ord('a')
            if left[idx] == -1: left[idx] = i
            right[idx] = i
        for i in range(26):
            if left[i] != -1 and right[i] != sz:
                chars = set(d[left[i] + 1:right[i]])
                ans += len(chars)
        return ans
